galgebra/printer.py

```diff
@@ -1129,3 +1130,66 @@ def Fmt(obj, fmt=0):
         return
     else:
         raise TypeError(str(type(obj)) + ' not allowed arg type in Fmt')
+
+
+class _WithSettings:
+    """ Helper class to attach print settings to an object """
+    def __init__(self, obj, settings: dict = {}):
+        self._obj = obj
+        self._settings = settings
+
+    def __do_print(self, printer):
+        # make a copy of the printer with the specified setting applied
+        new_printer = copy.copy(printer)
+        new_printer._settings = copy.copy(new_printer._settings)
+        new_printer._settings.update(self._settings)
+        return new_printer.doprint(self._obj)
+
+    _latex = _pretty = _sympystr = __do_print
```

Comment:
Supporting `_pretty` on `_FmtResult` would be fairly straightforward, but it's not useful unless we also implement it on `Mv`, `Lt`, ... etc